frame/nfts/src/functions.rs

```diff
@@ -130,6 +134,18 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			);
 			ensure!(collection_details.attributes == witness.attributes, Error::<T, I>::BadWitness);
 
+			for (account, roles) in CollectionRoleOf::<T, I>::drain_prefix(&collection) {
+				let roles = roles.values();
+				if roles.contains(CollectionRole::Admin) {
+					ensure!(account == witness.admin, Error::<T, I>::BadWitness);
+				}
+				if roles.contains(CollectionRole::Freezer) {
+					ensure!(account == witness.freezer, Error::<T, I>::BadWitness);
+				}
+				if roles.contains(CollectionRole::Issuer) {
+					ensure!(account == witness.issuer, Error::<T, I>::BadWitness);
+				}
```

Comment:
why the client of the call must provide accounts with the roles to destroy a collection?

Reply:
Very good question :D 
Initially I had another implementation in my head where these params were required, but then I came up with a better structure